tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -320,15 +321,24 @@ class ApplicationInfo(
     if (this.appStart.nonEmpty) {
       val appStartNew: ArrayBuffer[ApplicationCase] = ArrayBuffer[ApplicationCase]()
       for (res <- this.appStart) {
-        val durationResult = ProfileUtils.OptionLongMinusLong(this.appEndTime, res.startTime)
+
+        val estimatedResult = Option(this.appEndTime.getOrElse {
+          logWarning("Application End Time is unknown, estimating based on job and sql end times!")
+          // estimate the app end with job or sql end times
+          val sqlEndTime = this.sqlEndTime.values.toSeq.sorted(Ordering[Long].reverse).head
```

Comment:
what if sqlEndTimes (or  jobEndTime) is empty? Let us use something like `headOption.getOrElse(0L)

we don't need sort to determine the max. `if (sqlEndtime.isEmpty) 0L else sqlEndTime.values.max`

Reply:
good catch